tools/jenkins/run_distribution.sh

```diff
@@ -66,26 +66,77 @@ if [ "$platform" == "linux" ]; then
 elif [ "$platform" == "macos" ]; then
 
   if [ "$dist_channel" == "homebrew" ]; then
-    which brew # TODO: for debug, can be removed later
-    brew list -l
-    dir=/tmp/homebrew-test-$language
-    rm -rf $dir
-    mkdir -p $dir
-    git clone https://github.com/Homebrew/homebrew.git $dir
-    cd $dir
-    # TODO: Uncomment these when the general structure of the script is verified
-    # PATH=$dir/bin:$PATH brew tap homebrew/dupes
-    # PATH=$dir/bin:$PATH brew install zlib
-    # PATH=$dir/bin:$PATH brew install openssl
-    # PATH=$dir/bin:$PATH brew tap grpc/grpc
-    # PATH=$dir/bin:$PATH brew install --without-python google-protobuf
-    # PATH=$dir/bin:$PATH brew install grpc
-    PATH=$dir/bin:$PATH brew list -l
+    # system installed homebrew, don't interfere
     brew list -l
+
+    # Set up temp directories for test installation of homebrew
+    brew_root=/tmp/homebrew-test-$language
+    rm -rf $brew_root
+    mkdir -p $brew_root
+    git clone https://github.com/Homebrew/homebrew.git $brew_root
+
+    # Install grpc via homebrew
+    #
+    # The temp $PATH env variable makes sure we are operating at the right copy of
+    # temp homebrew installation, and do not interfere with the system's main brew
+    # installation.
+    #
+    # It's a bit non-ideal as we duplicate most of homebrew-grpc/script/install
+    # logic here but as the Jenkins macos instance has some different user/root
+    # privileges so spelling things out here might be cleaner.
+    cd $brew_root
+    PATH=$brew_root/bin:$PATH brew tap homebrew/dupes
+    PATH=$brew_root/bin:$PATH brew install zlib
+    PATH=$brew_root/bin:$PATH brew install openssl
+    PATH=$brew_root/bin:$PATH brew tap grpc/grpc
+    PATH=$brew_root/bin:$PATH brew install --without-python google-protobuf
+    PATH=$brew_root/bin:$PATH brew install grpc
+    PATH=$brew_root/bin:$PATH brew list -l
+
+    # Install per-language modules/extensions on top of core grpc
+    #
+    # If a command below needs root access, the binary had been added to
+    # /etc/sudoers. This step needs to be repeated if we add more mac instances
+    # to our jenkins project.
+    #
+    # Examples (lines that needed to be added to /etc/sudoers):
+    # + Defaults        env_keep += "CFLAGS CXXFLAGS LDFLAGS enable_grpc"
+    # + jenkinsnode1 ALL=(ALL) NOPASSWD: /usr/bin/pecl, /usr/local/bin/pip,
+    # +   /usr/local/bin/npm
+    case $language in
+      *core*) ;;
+      *python*)
+        sudo CFLAGS=-I$brew_root/include LDFLAGS=-L$brew_root/lib pip install grpcio
+        pip list | grep grpc
+        echo 'y' | sudo pip uninstall grpcio
+        ;;
+      *nodejs*)
+        sudo CXXFLAGS=-I$brew_root/include LDFLAGS=-L$brew_root/lib npm install --global grpc
```

Comment:
It still doesn't work, same error. There is indeed `pkg-config` under the `$brew_root/bin` directory.

```
Jenkinss-Mac-mini:~ jenkinsnode1$ echo $brew_root
/tmp/homebrew-test-temp
Jenkinss-Mac-mini:~ jenkinsnode1$ sudo PATH=$brew_root/bin:$PATH npm install grpc
-
> grpc@0.10.0 install /Users/jenkinsnode1/node_modules/grpc
> node-gyp rebuild
  CXX(target) Release/obj.target/grpc/ext/byte_buffer.o
../ext/byte_buffer.cc:38:10: fatal error: 'grpc/grpc.h' file not found
#include "grpc/grpc.h"
         ^
1 error generated.
make: *** [Release/obj.target/grpc/ext/byte_buffer.o] Error 1
```
Also tried `export PATH=$brew_root/bin:$PATH` first.

Perhaps the `grpc.pc` file has some problem?

```
Jenkinss-Mac-mini:~ jenkinsnode1$ pkg-config --list-all
protobuf-lite         Protocol Buffers - Google's Data Interchange Format
libpng16              libpng - Loads and saves PNG files
gio-unix-2.0          GIO unix specific APIs - unix specific headers for glib I/O library
gio-2.0               GIO - glib I/O library
libpcreposix          libpcreposix - PCREPosix - Posix compatible interface to libpcre
protobuf              Protocol Buffers - Google's Data Interchange Format
Variable 'prefix' not defined in '/private/tmp/homebrew-test-temp/lib/pkgconfig/grpc.pc'
```